nemo/collections/llm/api.py

```diff
@@ -716,3 +719,99 @@ def _set_with_io(obj, attr, value):
     setattr(obj, attr, value)
     if hasattr(obj, "__io__") and hasattr(value, "__io__"):
         setattr(obj.__io__, attr, deepcopy(value.__io__))
+
+
+def _validate_config(
+    model: pl.LightningModule,
+    data: pl.LightningDataModule,
+    trainer: Trainer,
+    log: Optional[NeMoLogger] = None,
+    resume: Optional[AutoResume] = None,
+    optim: Optional[OptimizerModule] = None,
+    tokenizer: Optional[TokenizerType] = None,
+    model_transform: Optional[Union[PEFT, ModelTransform, Callable]] = None,
+) -> None:
+
+    ## Model validation
+    assert getattr(model.config, "seq_length", 1) > 0
+    assert getattr(model.config, "max_position_embeddings", 1) > 0
+    assert model.config.num_layers > 0
+    assert model.config.hidden_size > 0
+    assert model.config.num_attention_heads > 0
+    assert model.config.ffn_hidden_size > 0
+
+    if hasattr(model.config, "seq_length"):
+        if getattr(model.config, "max_position_embeddings", None) is not None:
+            assert model.config.seq_length <= model.config.max_position_embeddings
+
+    ## Data validation
+    assert data.micro_batch_size > 0
+    assert data.global_batch_size > 0
+    assert data.seq_length > 0
+
+    assert (
+        data.global_batch_size % data.micro_batch_size == 0
+    ), "Global batch size must be divisible by micro batch size in data module."
+
+    ## Trainer validation
+
+    # Basic validation
+    assert trainer.strategy.tensor_model_parallel_size > 0
+    assert trainer.strategy.pipeline_model_parallel_size > 0
+    assert trainer.strategy.context_parallel_size > 0
+
+    # DP validation
```

Comment:
You are assuming here that a user always uses `MegatronStrategy`, that's not the case. So please add some if-statements in this function

Reply:
Thanks for pointing out. I have fixed this part.